typecheck/typecheck_test.go

```diff
@@ -94,12 +94,13 @@ var typeTests = []typeTest{
 			{"b", tipe.Float64},
 		},
 	},
-	{
-		[]string{
-			`a := [|]int64{{|"Col1","Col2"|}, {1, 2}, {3, 4}}`,
-		},
-		[]identType{{"a", &tipe.Table{tipe.Int64}}},
-	},
+	//	TODO:
```

Comment:
What happened here? Was the test never working? I don't see how this change could have broken it.

Reply:
by correctly handling the binary `|` operator, it did break parsing `{|` and `|}`.
I do have in [sbinet/tableliteral](https://github.com/neugram/ng/commit/7d795a6c6efc763b8f51f85c275c3992d547d457) a change that reenables this test.

I introduced 2 new tokens: `{|` and `|}`